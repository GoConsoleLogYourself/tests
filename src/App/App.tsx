// import { useEffect, useState } from "react";
// import "./App.css";

// function App() {
//   const [data, setData] = useState<number>();
//   const [toggle, setToggle] = useState<boolean>(false);
//   const [value, setValue] = useState<string>("");

//   const onClick = () => {
//     setToggle((prev) => !prev);
//   };
//   useEffect(() => {
//     setTimeout(() => {
//       setData(5);
//     }, 100);
//   }, []);
//   return (
//     <div>
//       <h1 data-testid="input-value">{value}</h1>
//       {toggle === true && <div data-testid="toggle-elem">Toggle</div>}
//       {data && <div>Data</div>}
//       <h1>Hello world</h1>
//       <button data-testid="button" onClick={onClick}>
//         Click me
//       </button>
//       <input
//         value={value}
//         onChange={(e) => setValue(e.target.value)}
//         type="text"
//         placeholder="input value"
//       />
//     </div>
//   );
// }

// export default App;

import AppRouter from "../routes/AppRouter";
import Navbar from "../components/navbar/Navbar";

const App = () => {
  return (
    <div>
      <Navbar />
      <AppRouter />
    </div>
  );
};

export default App;
