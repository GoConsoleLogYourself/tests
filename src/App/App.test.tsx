import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("TEST APP", () => {
  test("renders", async () => {
    render(<App />);
    // const helloWorldElem = screen.queryByText(/Hello world2/i);
    // expect(helloWorldElem).toBeNull()
    screen.debug();
    const helloWorldElem = await screen.findByText(/Data/i);
    expect(helloWorldElem).toBeInTheDocument();
    screen.debug();
  });

  test("CLICK EVENT", () => {
    render(<App />);
    const btn = screen.getByTestId("button");
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId("toggle-elem")).toBeNull();
  });

  test("INPUT EVENT", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value/i);
    expect(screen.queryByTestId("input-value")).toContainHTML("");
    fireEvent.input(input, {
      target: { value: "1234" },
    });
    // userEvent.type(input, "1234");
    expect(screen.queryByTestId("input-value")).toContainHTML("1234");
  });
});
