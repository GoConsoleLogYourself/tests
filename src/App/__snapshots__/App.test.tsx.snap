// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TEST APP > renders 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
